packages/react/src/components/IconButton/index.tsx

```diff
@@ -5,16 +5,99 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import PropTypes from 'prop-types';
-import React from 'react';
-import Button from '../Button';
+import PropTypes, { ReactNodeLike } from 'prop-types';
+import React, { ForwardedRef } from 'react';
+import Button, { ButtonSize, ButtonKind } from '../Button';
 import classNames from 'classnames';
 import { Tooltip } from '../Tooltip';
 import { usePrefix } from '../../internal/usePrefix';
 import cx from 'classnames';
 
-const IconButton = React.forwardRef(function IconButton(props, ref) {
-  const {
+interface IconButtonProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * Specify how the trigger should align with the tooltip
+   */
+  align?:
```

Comment:
Seems like this `align` is missing `left-bottom`, `left-top`, `right-bottom`, `right-top`. Based on the underlying `Tooltip`, we should be able to pull these in from `Popover`


```jsx
import { PopoverAlignment } from '../Popover';
  /**
   * Specify how the trigger should align with the tooltip
   */
  align?: PopoverAlignment;
```

Reply:
@tw15egan Do you know if that is intentional or an actual oversight? The proptype below doesn't include these and is only: 
```
align: PropTypes.oneOf([
    'top',
    'top-left',
    'top-right',
    'bottom',
    'bottom-left',
    'bottom-right',
    'left',
    'right',
  ]),
```

@laurenmrice is the iconbutton limited to only certain alignment options for the tooltip?

Reply:
@tay1orjones seems like an oversight, not sure why we wouldn't have those two extra alignments for `left` and `right`?

Reply:
@tw15egan I'm not sure why, but back in the original PR those alignments were explicitly not included. https://github.com/carbon-design-system/carbon/pull/10095#pullrequestreview-807656822

Reply:
Just confirmed we likely didn't have all of them at first because IconButton tooltips are usually only a word or two and might not need all the extra alignments.

Lauren [mentioned](https://ibm-studios.slack.com/archives/GD8JG6JR3/p1699303094236079) we can add them all in though, just in case people need them. I'll update the PR.